food-truck-api/src/main/java/food/truck/api/endpoint/TruckEndpoint.java

```diff
@@ -181,15 +202,16 @@ public boolean updateRoute(@AuthenticationPrincipal User u, @PathVariable long t
     }
 
     @PostMapping("/truck/{truckId}/create-route")
+    @Secured("ROLE_OWNER")
     public Route createTruckRoute(@AuthenticationPrincipal User user, @PathVariable long truckId, @RequestBody PostRouteParams data) {
-        Optional<Truck> truck = truckService.findTruckById(truckId);
-        if (truck.isEmpty()) {
-            return new Route();
-        }
-        return routeService.createRoute(truck.get(), data.routeName, Character.toUpperCase(data.active) == 'Y'); // TODO
+        if (!truckService.userOwnsTruck(user, truckId))
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN);
+
+        return routeService.createRoute(truckService.findTruckById(truckId).get(), data.routeName, Character.toUpperCase(data.active) == 'Y');
```

Comment:
It's possible that truck might return as none, so it would probably be better to check and not create a route when there isn't a truck.

Reply:
That's checked in userOwnsTruck, but good eye